File: src/compiler/tokenizer.rs
use crate::utils::is_delim;

fn pad_delims(str: &str) -> String {
    str.chars()
        .flat_map(|c| {
            if is_delim(c) {
                vec![' ', c, ' ']
            } else {
                vec![c]
            }
        })
        .collect()
}

pub fn tokenize(code: &str) -> Vec<String> {
    pad_delims(code)
        .split_whitespace()
        .map(String::from)
        .collect()
}

File: lib/compiler/tokenizer.dart
import 'package:bliss/utils.dart';

final _whitespaceRegex = RegExp("\\s");

String _padDelims(String string) {
  return string.split("").map((c) => isDelim(c) ? " $c " : c).join();
}

List<String> tokenize(String code) {
  return _padDelims(
    code,
  ).split(_whitespaceRegex).where((x) => x.isNotEmpty).toList();
}
